projects/js-toolkit/packages/js-api/editor/index.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+/**
+ * Editor Config Transformer
+ *
+ * For CKEditor 4, see API of CKEDITOR.config:
+ * https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html
+ */
+
+export interface EditorConfig {
+	[key: string]: any;
+}
+
+export interface EditorConfigTransformer {
+	(args: EditorConfig): EditorConfig;
```

Comment:
Also, I would consider exporting an object instead of a raw function (from the CX impl) unless you are 200% :sweat_smile:  sure users will never need to export metadata related to the function or even more functions.

Remember that I had to change that in `FDSFilter` when I added more functions to the CX as a consequence of evolving the CX features. So given that returning the function inside an object instead of returning it directly is not very cumbersome for the user and allows for much more future flexibility, I think we should consider it.

Reply:
Hmm the scope of customization that we always considered is the `config` object. This matches `EditorConfigContributor`s in DXP, through which we can only modify the `config` object as well. As long as that is the scope, I'd be 200% confident we will never need to export more. But! There is a lot more to [CKEditor API](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR.html) other than config object. You can modify and override its implementation, like we are [doing in plugins](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/ballooneditor/plugin.js#L231). If we want to be open to support those kinds of customizations later on, it would make sense to set an object return value immediately, so it's compatible later on. In that case, this would better be named something like `EditorTransformer`.